sdk/common/src/main/java/io/opentelemetry/sdk/common/export/CompletableResultCode.java

```diff
@@ -107,4 +109,33 @@ public CompletableResultCode whenComplete(Runnable action) {
     }
     return this;
   }
+
+  /** Waits until this instances completes, regardless of the result. */
+  public void join() throws InterruptedException {
+    final CountDownLatch latch = new CountDownLatch(1);
+    whenComplete(
+        new Runnable() {
+          @Override
+          public void run() {
+            latch.countDown();
+          }
+        });
+    latch.await();
+  }
+
+  /** Calls {@link #fail()} if times out. */
+  public void join(long timeout, TimeUnit unit) throws InterruptedException {
```

Comment:
`CompletableFuture.join` doesn't throw interrupted exception so I think it's ok to squash that here too (don't know if it's best practice for interruptions but may as well copy the JDK)

Reply:
You mean catch that inside the method?